### PR TITLE
Hotfix request level cache

### PR DESCRIPTION
**Pull request by toplenboren**
This is a work in progress try to catch a wandering bug with `cache.js` module. Bug is caused (well, we think it caused) by race-condition.

### CODE REVIEW

File: packages/keystone/cache.js [hunk 1]
```diff
@@ -97,7 +99,7 @@ const patchQuery = (queryContext, query, cache) => {
         }
         const listResult = await query.call(queryContext, args, context, gqlName, info, from)
 
-        if (requestId && key) { cache[requestId][key] = listResult }
+        if (requestId && key && get(cache, [requestId, key])) { cache[requestId][key] = listResult }
```
Comment (pahaz) [inline packages/keystone/cache.js:102]: probably it's easy to use lodash set?

File: apps/condo/domains/user/schema/User.test.js [hunk 2]
```diff
@@ -415,3 +415,72 @@ describe('UserHistoryRecord', () => {
         ])
     })
 })
+
+describe('Cache tests', () => {
+    test('Clients that ask for different set of fields get different set of fields', async () => {
```
Comment (pahaz) [inline apps/condo/domains/user/schema/User.test.js:420]: it looks ledundant

File: packages/keystone/cache.js (hunk 1 above)
Comment (toplenboren) [reply packages/keystone/cache.js]: Done!

File: apps/condo/domains/user/schema/User.test.js (hunk 2 above)
Comment (toplenboren) [reply apps/condo/domains/user/schema/User.test.js:420]: Do we need to remove it?

**Review by pahaz — APPROVED** [review]

**Review by pahaz — APPROVED** [review]

**Pull request merged**